Use move semantics for ShareableBitmap::create

donny-dont opened the pull request:
#### 802c13cdf46a0eb2c1f86960926c45885793132b
<pre>
Use move semantics for ShareableBitmap::create
<a href="https://bugs.webkit.org/show_bug.cgi?id=256112">https://bugs.webkit.org/show_bug.cgi?id=256112</a>

Reviewed by Kimmo Kinnunen.

Continuation of 263363@main which used move semantics for
`ShareableBitmap::Handle` to signify transferring ownership.

* Source/WebKit/Shared/ShareableBitmap.cpp:
* Source/WebKit/Shared/ShareableBitmap.h:
* Source/WebKit/Shared/gtk/ArgumentCodersGtk.cpp:
* Source/WebKit/UIProcess/BackingStore.cpp:

Canonical link: <a href="https://commits.webkit.org/263689@main">https://commits.webkit.org/263689@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c792e75a8ebcf244815ce291c9854c900f9c9bd4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/5422 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/5561 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5753 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6964 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/5450 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5790 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/5540 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/6359 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/5524 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/5588 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4857 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6993 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3066 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/12016 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4932 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4942 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/6648 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/5377 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/4414 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4837 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1302 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8938 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5200 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple commented on Source/WebKit/Shared/UpdateInfo.h:
```diff
@@ -68,7 +68,7 @@ class UpdateInfo {
 
     // The handle of the shareable bitmap containing the updates. Will be null if there are no updates.
-    ShareableBitmap::Handle bitmapHandle;
+    mutable ShareableBitmap::Handle bitmapHandle;
```
I think, here is the same problem:
Here you make this mutable, and there's a const UpdateInfo& receiver that then clobbers the instance and makes the instance "it kind of is ok, but it really is broken as the bitmapHandle was moved away".

Instead of making this mutable, you need to make all `const UpdateInfo&` receivers as `UpdateInfo&&`.

donny-dont replied:
https://github.com/WebKit/WebKit/pull/13418 Addressed there

kkinnunen-apple reviewed (commented):
Since we're doing this for the sake of going "by the book", this issue should be included

donny-dont committed:
Use move semantics for ShareableBitmap::create
https://bugs.webkit.org/show_bug.cgi?id=256112

Reviewed by Kimmo Kinnunen.

Continuation of 263363@main which used move semantics for
`ShareableBitmap::Handle` to signify transferring ownership.

* Source/WebKit/Shared/ShareableBitmap.cpp:
* Source/WebKit/Shared/ShareableBitmap.h:
* Source/WebKit/Shared/gtk/ArgumentCodersGtk.cpp:
* Source/WebKit/UIProcess/BackingStore.cpp:

Canonical link: https://commits.webkit.org/263689@main
